lib/plugins/stripTrailingSlashPlugin.ts

```diff
@@ -3,9 +3,9 @@ import fp from 'fastify-plugin'
 
 const pluginCallback: FastifyPluginCallback = (fastify, _options, done) => {
   fastify.addHook('onRequest', (request, reply, done) => {
-    const { pathname, search } = new URL(
-      `${request.protocol}://${request.hostname}:${request.port}${request.url}`,
-    )
+    // Using a hardcoded origin for simplicity and reliability,
+    // as the request's origin is not relevant in this case.
+    const { pathname, search } = new URL(request.url, 'https://example.com')
```

Comment:
can you also add a new test to ensure whatever case you are beefing up against is covered?

Reply:
We're not able to invoke local fastify app without specifying a port. I've changed the implementation, so the passed port doesn't matter.